site/book/07-effective-customizations/01-single-value-replacement.md

```diff
@@ -0,0 +1,26 @@
+## Scenario
+
+I have a single value replacement in my package, I don’t want package consumers to look through all the yaml files to find the value I want them to set, it seems easier to just create a parameter for this value and have the user look at Kptfile for inputs.
+
+## Problems
+
+1. With popularity the single values inevitably expand to provide a facade to a large portion of the data defeating the purpose of minimizing the cognitive load.
+1. Some values like resource names are used as references so setting them in one place needs to trigger updates in all the places where they are referenced.
+1. If additional resources that have similar values are added to the package new string replacements need to be added.
+1. If a package is used as a sub-package the string replacement parameters need to be surfaced to the parent package and if the parent package already expects some values to be set and the parameters do not exist, the sub-package needs to be updated.
+
+## Suggestions
+
+1. kpt allows the user to edit a particular value directly in the configuration data and will handle upstream merge.  When [editing the yaml] directly the consumers are not confined to the parameters that the package author has 
```

Comment:
We could also explain this as similar to kustomize but where the patch is automatically derived